pymatgen/io/zeopp.py

```diff
@@ -117,7 +117,8 @@ def from_str(cls, string: str) -> Self:
         lengths.insert(0, a)
         alpha = angles.pop(-1)
         angles.insert(0, alpha)
-        lattice = Lattice.from_parameters(*lengths, *angles)
+        lattice = Lattice.from_parameters(*lengths, *angles)  # type: ignore[arg-type]
```

Comment:
Not sure if there is a better way out. We're having multiple `mypy` complains related to unpacking args.

Reply:
Last unresolved here @janosh.

Sorry I'm making this PR too large to navigate.

Reply:
> Sorry I'm making this PR too large to navigate.

all good! there's a lot of legacy stuff to clean up in `pymatgen` mkaing large PRs necessary

[i think i found a cleaner solution to this problem](https://github.com/materialsproject/pymatgen/pull/3705/commits/6d06bef4eb10a9c1e3062c94fa8001757b686d39#diff-5215f3d451242a8bcb7a19dde68e5afa7583bafc7a598039da82f03f9fd91c77R322)

Reply:
Yes [I did notice that](https://github.com/materialsproject/pymatgen/pull/3705#discussion_r1544278631). But frankly I have never seen such usage before. Hopefully `mypy` could have a "smarter" way to detect this in the near future (otherwise we would need to do the same for all unpacking of args).